src/common/cockpitwebserver.c

```diff
@@ -1075,6 +1052,17 @@ on_request_input (GObject *pollable_input,
       return FALSE;
     }
 
+  length += count;
+  g_byte_array_set_size (request->buffer, length);
```

Comment:
There is a potential gotcha here: It could theoretically be that the buffer gets reallocated to fit the current size instead of 8 KiB. (That would be stupid to do in practice, but it's a new assumption). To pin this down, we could either put another `g_byte_array_set_size(cockpit_webserver_request_maximum)` at the top, or just directly set `buffer->len` instead of calling `set_size()`. The latter feels a bit hackish, though. Any opinions?